protos/backend_service.proto

```diff
@@ -168,4 +171,47 @@ message PingRequest {
 // Response payload for ping operations.
 message PingResponse {
 	// No fields
-}
\ No newline at end of file
+}
+
+// Request payload for fetching backend metrics.
+message GetMetricsRequest {
+    // No fields
+}
+
+// Response payload for fetching backend metrics
+message GetMetricsResponse {
+    // The current metrics for the backend service.
+    BackendMetrics metrics = 1;
+}
+
+// Metrics for the backend service.
+message BackendMetrics {
+    // Activity work item metrics
+    WorkItemMetrics activityWorkItems = 1;
+    // Orchestrator work item metrics
+    WorkItemMetrics orchestratorWorkItems = 2;
+    // Entity work item metrics
+    WorkItemMetrics entityWorkItems = 3;
+    // Metrics related to workers currently connected to the backend
+    ConnectedWorkerMetrics connectedWorkers = 4;
+}
+
+// Metrics related to work items
+message WorkItemMetrics {
+    // Number of work items that are queued and waiting to be processed
+    int32 pending = 1;
```

Comment:
Out of curiosity, why int32 rather than int64? (or uint64)

Reply:
I don't expect we'll ever have anything close to 2 billion pending work items in a single task hub.

Reply:
Fair.

If we wanted to change this in the future, it's backwards-compatible anyways (in the wire), since protobuf encodes all ints as varints.